src/transformers/generation/candidate_generator.py

```diff
@@ -450,9 +449,9 @@ def get_candidates(self, input_ids: torch.LongTensor) -> Tuple[torch.LongTensor,
         # Since re-encoding the tokens may result in tokenization discrepancies, we use 2 look behind values
         # (one for each conversion) which mark where to start looking for the overlap between the
         # source and target encodings, to ensure the new tokens include the correct prompt suffix.
-        if self.prev_tokens is not None and self.prev_target_ids.shape[1] > self.target_lookbehind:
+        if self.prev_assistant_ids is not None and input_ids.shape[1] > self.target_lookbehind:
             # input_ids contains all target prompt input ids and some new target input ids
-            start_index_in_target_window = self.prev_target_ids.shape[1] - self.target_lookbehind
+            start_index_in_target_window = input_ids.shape[1] - self.target_lookbehind
```

Comment:
isn't this different from current impl because `input_ids` are not same as the previous `input_ids`?

Reply:
Yes. 
`self.prev_target_ids` contains the previous `input_ids` and candidate tokens (have been converted by target tokenizer), but not all candidate tokens are accepted by the target model. So the number of `self.prev_target_ids` is not equal to the number of `input_ids` in the current step.

Actually, we can directly use the `inpud_ids` to get the correct length of tokens, and there is no need to save config `self.prev_target_ids`.

Reply:
Yes, it is.

Reply:
cool, requesting core maintainer review and then we can merge